### PR TITLE
Faster windoors

### PR DESCRIPTION
**Pull request by imsynthex**
## About The Pull Request

Speeds windoors opening and closing up 2x

Makes the jaws of life work on windoors

## Why It's Good For The Game

Current windoors are obnoxiously slow when compared to regular doors, seeing as they're used a lot for applications that require you to walk through them (such as HoP and captain's offices on metastation, as seen on the video), it makes sense for them to be faster as to not get in the way as much

## Testing Photographs and Procedure

https://user-images.githubusercontent.com/81387903/222989212-94a7cbd9-edd2-416a-a624-3418c693bbea.mp4

## Changelog
:cl:
tweak: windoors are now twice as fast to open and close
balance: jaws of life now work on windoors
/:cl:

### CODE REVIEW

**Review by PowerfulBacon — DISMISSED** [review]
The animation needs fixing, in the video it opens half way before teleporting open

**Comment by imsynthex** [comment]
> The animation needs fixing, in the video it opens half way before teleporting open

Changed the animation to be smoother and added a new video (this time with audio) to the PR body

**Comment by Tsar-Salat** [comment]
Currently, windoors don't respect Jaws of Life, which doesnt make sense that it takes more effort to open a glass pane than a airlock with 1k ibs of pressure behind it.

Can you implement:
- https://github.com/tgstation/tgstation/pull/71733

If you think it would fit with your PR.

**Comment by imsynthex** [comment]
> Currently, windoors don't respect Jaws of Life, which doesnt make sense that it takes more effort to open a glass pane than a airlock with 1k ibs of pressure behind it.
> 
> If you think it would fit with your PR.

For whatever reason porting it didn't work, not going to look into it, just made another implementation for the same thing

**Comment by Tsar-Salat** [comment]
Very nice

File: code/game/machinery/doors/windowdoor.dm [hunk 1]
```diff
@@ -72,25 +73,25 @@
 		close()
 
 /obj/machinery/door/window/Bumped(atom/movable/AM)
-	if( operating || !density )
+	if(operating || !density)
 		return
-	if (!( ismob(AM) ))
+	if(!(ismob(AM)))
```
Comment (itsmeow) [inline code/game/machinery/doors/windowdoor.dm]: unnecessary parenthesis around ismob

File: code/game/machinery/doors/windowdoor.dm [hunk 2]
```diff
@@ -72,25 +73,25 @@
...
 		if(ismecha(AM))
 			var/obj/mecha/mecha = AM
 			if(mecha.occupant && allowed(mecha.occupant))
 				open_and_close()
 			else
 				do_animate("deny")
 		return
-	if (!( SSticker ))
+	if(!(SSticker))
```
Comment (itsmeow) [inline code/game/machinery/doors/windowdoor.dm]: unnecessary parenthesis again. also SSticker can't be null lmao, delete this tomfoolery

File: code/game/machinery/doors/windowdoor.dm [hunk 3]
```diff
@@ -301,11 +302,11 @@
 	try_to_activate_door(null, user)
 
 /obj/machinery/door/window/try_to_activate_door(obj/item/I, mob/user)
-	if (..())
+	if(..())
 		autoclose = FALSE
 
 /obj/machinery/door/window/try_to_crowbar(obj/item/I, mob/user)
-	if(!hasPower())
+	if(!hasPower() || HAS_TRAIT(I, TRAIT_DOOR_PRYER))
```
Comment (itsmeow) [inline code/game/machinery/doors/windowdoor.dm]: Door prying for regular airlocks has a do_after that takes like 5 seconds, it should do the same

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Review by itsmeow — APPROVED** [review]

**Pull request merged**